File: lib/db/models/arma_3_server.py
from lib.db.models.model_deletion_logic import MongoModelDeletionLogic
from lib.settings import Settings
import os
import logging

logger = logging.getLogger(__name__)


class Arma3Server(MongoModelDeletionLogic):
    @classmethod
    def _get_collection_name(cls):
        return 'arma_3_servers'

    def _get_fields(self):
        return [
            'name',
            'port',
            'modset_id',
            'additional_commandline',
        ]

    @classmethod
    def _delete_from_record(cls, record):
        server_id = record['id']

        if Settings.debug_windows:
            logger.info("windows delete server dummy")
            return

File: MoonlightDashboard/MoonlightDashboard/Database/Models/Arma3Server.cs
namespace MoonlightDashboard.Database.Models
{
    public class Arma3Server : Model
    {
        public required string Name { get; set; }
        public required int Port { get; set; }
        public string? ActiveModsetId { get; set; }
        public string AdditionalCommandlineArgs { get; set; } = "";
    }
}
